tutorials/securing-gcs-static-website/index.md

```diff
@@ -0,0 +1,455 @@
+
+### Deploy login service to Cloud Run
+
+1. Build the Docker container for the login page and push it to Container Registry.
+
+    ```bash
+    cd ../flask_login
+
+    docker build -t flask_login .
+    docker tag flask_login gcr.io/$PROJECT_ID/flask_login
+    docker push gcr.io/$PROJECT_ID/flask_login
+    ```
+
+1. Deploy Cloud Run service. For demo purposes, we pass the user credential through environment variables. In reality, you probably want to use a user database or an identity provider for the login service
+
+    ```bash
+    gcloud run deploy $LOGIN_BACKEND_SVC_NAME \
+    --image=gcr.io/$PROJECT_ID/flask_login --platform=managed \
+    --region=$REGION --allow-unauthenticated \
+    --set-env-vars=WEB_URL=https://$DNS_NAME,PROJECT_ID=$PROJECT_ID,CDN_SIGN_KEY=$CDN_SIGN_KEY,USER_NAME=admin,USER_PASSWORD=password
+    ```
+
+1. Since our Cloud Run service needs to access the secrets saved in the secret manager, we grant the permission here. 
+
+    ```bash
+    gcloud projects add-iam-policy-binding \
+    --member=serviceAccount:${PROJECT_NUM}-compute@developer.gserviceaccount.com \
+    --role=roles/secretmanager.secretAccessor $PROJECT_ID
+    ```
+
+    __Note__: we provide the secretAccessor permission to the default compute service account. In a production environment, you probably want to use a custom service account and only allow access to the needed secrets.
+
+### Configure serverless network endpoints group
+
+1. Create a network endpoint group(NEG) for the Cloud Run service.
+
+    ```bash
+    gcloud beta compute network-endpoint-groups create $SERVERLESS_NEG_NAME \
+        --region=$REGION \
+        --network-endpoint-type=SERVERLESS  \
+        --cloud-run-service=$LOGIN_BACKEND_SVC_NAME
+    ```
+
+1. Create a backend service and add the serverless NEG as a backend to the Cloud Run service. A serverless NEG is needed here because that's how Cloud Run services can be associated with a load balancer.
+
+    ```bash
+    gcloud compute backend-services create $LOGIN_BACKEND_SVC_NAME \
+        --global
+
+    gcloud beta compute backend-services add-backend $LOGIN_BACKEND_SVC_NAME \
+        --global \
+        --network-endpoint-group=$SERVERLESS_NEG_NAME \
+        --network-endpoint-group-region=$REGION
+    ```
+
+### Create URL map and configure forwarding rules
+
+1. Update the URL mapping file and create the URL map.
+
+    ```bash
+    sed -i -e "s/<DNS_NAME>/$DNS_NAME/" web-map-http.yaml
+    sed -i -e "s/<PROJECT_ID>/$PROJECT_ID/" web-map-http.yaml
+    sed -i -e "s/<LOGIN_BACKEND_SVC_NAME>/$LOGIN_BACKEND_SVC_NAME/" web-map-http.yaml
+
+    gcloud compute url-maps import web-map-http --source web-map-http.yaml --global
+    ```
+
+    In this step, we updated the values in the template URL map file and imported it. An example of the final configuration looks like the following:
```

Comment:
This IS the final config, not really an example of

Reply:
removed the word 'example'